sdmetrics/single_table/new_row_synthesis.py

```diff
@@ -70,30 +71,36 @@ def compute_breakdown(cls, real_data, synthetic_data, metadata=None,
             else:
                 synthetic_data = synthetic_data.sample(n=synthetic_sample_size)
 
-        numerical_fields = []
-
         for field, field_meta in get_columns_from_metadata(metadata).items():
-            field_type = get_type_from_column_meta(field_meta)
-
-            if field_type == 'datetime':
+            if get_type_from_column_meta(field_meta) == 'datetime':
                 real_data[field] = pd.to_numeric(real_data[field])
                 synthetic_data[field] = pd.to_numeric(synthetic_data[field])
-                numerical_fields.append(field)
-            elif field_type == 'numerical':
-                numerical_fields.append(field)
+
+        try:
+            numerical_fields = cls._select_fields(metadata, ('numerical', 'datetime'))
+        except IncomputableMetricError:
+            numerical_fields = []
+
+        try:
+            categorical_fields = cls._select_fields(metadata, ('categorical', 'boolean'))
+        except IncomputableMetricError:
+            categorical_fields = []
```

Comment:
very minor: but you could wrap these try/excepts in a method that gets the fields of a type, catches the error and returns an empty list if it does.